source/common/http/conn_manager_impl.cc

```diff
@@ -2079,6 +2079,20 @@ void ConnectionManagerImpl::FilterManager::encodeTrailers(ActiveStreamEncoderFil
     }
   }
 
+  // Some browsers (e.g. WebKit-based browsers: https://bugs.webkit.org/show_bug.cgi?id=210108) have
+  // a problem with processing empty trailers (END_STREAM | END_HEADERS with zero length HEADERS) of
+  // an HTTP/2 response as reported here: https://github.com/envoyproxy/envoy/issues/10514.
+  const bool skip_encoding_empty_trailers =
+      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.skip_encoding_empty_trailers");
```

Comment:
If it's just HTTP/2 should we do it in the codec?  It would potentially affect the request and response, but it'd leave complexity outside the HCM and wouldn't affect HTTP/1.1 with trailers.  I'd mildly prefer that location but I'm Ok doing it here if we explicitly check that it's HTTP/2

Reply:
+1 to move to the codec

Reply:
Moved to codec.